File: A. Easy/0941-valid-mountain-array/0941-valid-mountain-array.c
bool validMountainArray(int* arr, int arrSize){
    if (arrSize < 3) {
        return false;
    }

    int i = 0, j = arrSize - 1;

    while ((i < (arrSize - 1)) && (arr[i] < arr[i + 1])) {
        i++;
    }

    while ((j > 0) && (arr[j] < arr[j - 1])) {
        j--;
    }

    return ((i==j) && (i != 0) && (j != (arrSize - 1)));
}

File: 2. Arrays 101/0941-valid-mountain-array/0941-valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n = len(arr)
        if n < 3:
            return False

        i, j = 0, n - 1

        # Ascend to the peak
        while i < n - 1 and arr[i] < arr[i + 1]:
            i += 1

        # Descend to the peak
        while j > 0 and arr[j] < arr[j - 1]:
            j -= 1

        # Check if i and j meet at the peak and are not at the start or end
        return i == j and i != 0 and j != n - 1
            